Add form fields for editing personal information

pcraig3 opened the pull request:
This pull request adds the ability to edit the first 4 fields: name, address, marital status, and number of children. (Note: adding the form elements themselves was merged as part of #33 and #39.)

It's a fairly naive implementation: editing is super simplified, but it meets the requirements of the ["Change information"](https://trello.com/c/cGCpHSIV/36-online-confirmation-information-change) story, and it's something we can build out further as needed. 

It's also pretty accessible, following many of the guidelines of the [GOV.UK design system](https://design-system.service.gov.uk/):

- [One thing per page](https://designnotes.blog.gov.uk/2015/07/03/one-thing-per-page/)
- [Text inputs](https://design-system.service.gov.uk/components/text-input/)
- [Textareas](https://design-system.service.gov.uk/components/textarea/)
- [Radio buttons](https://design-system.service.gov.uk/components/radios/)
- [Error messages](https://design-system.service.gov.uk/components/error-message/)
- [Error summaries](https://design-system.service.gov.uk/components/error-summary/)

I'm sure this will change as we move forwards and find out more about what values we are likely to be able to change, but it's simple enough to start with and gets us going in the right direction.

## gif

![forms](https://user-images.githubusercontent.com/2454380/56738200-1c2d3680-673a-11e9-97d7-151903f54be0.gif)

pcraig3 committed:
Add edit page for name

I don't have all the other form fields in the app yet, but all of
this so far is pretty generalizable.

Ideally, this one /edit/:id route will be able to handle editing
all of the fields we have.

pcraig3 committed:
Add label text to question

This way, our label text can be a bit more verbose.
ids and label text serve different purposes.

pcraig3 committed:
Allow address field to be edited

Using a regular expression in the route parameter so that only
two `:id`s are allowed: 'name' or 'address'.

When one of these is called, the appropriate form field is pulled
up with all the content related to the question.

I'm also keeping validation schemas in there, so we can also
validate that the answers that come back are what we expect.

Address and name specifically are going to have to change one day,
but it's a nice idea, and, further, I think many of the simpler fields
will fit this model. (Hopefully, anyway.)

pcraig3 committed:
Add back link to question schema

pcraig3 committed:
Add maritalStatus edit page

pcraig3 committed:
Add marital status question file

This is the first one that has "options", and I have to reference
them twice so I pulled them out into a separate variable.

The syntax for schema definition includes a nested array, which is
pretty unintuitive, but it works.

References:
- https://express-validator.github.io/docs/schema-validation.html
- https://github.com/express-validator/express-validator/issues/588

pcraig3 committed:
Add in children question file

We have to make sure it's a number, and there's even an input type
called "number" so let's get that going.

pcraig3 committed:
Allow editing the number of children field

Same as a regular input, but I made the field a lot shorter because
we want to imply that you shouldn't put a large value in there.

pcraig3 committed:
Make the page title on edit pages match the <h1>

pcraig3 committed:
Add validation error and error styling to Radios fieldset

Took example HTML from the GOV.UK Design Patterns, once again.

Also updated the negative margins + padding situation when we have
validation errors to take account of the border width.

References:
- https://design-system.service.gov.uk/components/radios/#error-messages

pcraig3 committed:
Trim whitespace from modified data

sboisvert commented on src/server.js:
```diff
@@ -132,6 +132,56 @@ app.get('/edit', checkLogin, (req, res) => {
   res.send(_renderDocument({ title: '[WIP] Edit', locale, content }))
 })
 
+app.get('/edit/:id(name|address|maritalStatus|children)?', checkLogin, (req, res) => {
```
We might want to add a comment pointing to https://expressjs.com/en/guide/routing.html and/or https://www.npmjs.com/package/path-to-regexp just so it's easier to know how this works

pcraig3 committed:
Add comment explaining our routing regex syntax

The official express documentation on this is pretty good.
- https://expressjs.com/en/guide/routing.html

But the stackoverflow answer linked to has this specific syntax.
- https://stackoverflow.com/a/15350845

Also remove the base /edit route. It was a placeholder and we're
not using it anyomre.
